pkg/apis/workflow/v1alpha1/http_types.go

```diff
@@ -43,6 +48,19 @@ type HTTP struct {
 	SuccessCondition string `json:"successCondition,omitempty" protobuf:"bytes,6,opt,name=successCondition"`
 	// Body is content of the HTTP Request
 	Body string `json:"body,omitempty" protobuf:"bytes,5,opt,name=body"`
-	// insecureSkipVerify is a bool when if set to true will skip TLS verification for the HTTP client
+	// BodyFrom is  content of the HTTP Request as Bytes
+	BodyFrom *HTTPBodySource `json:"bodyFrom,omitempty" protobuf:"bytes,8,opt,name=bodyFrom"`
+	// InsecureSkipVerify is a bool when if set to true will skip TLS verification for the HTTP client
 	InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty" protobuf:"bytes,7,opt,name=insecureSkipVerify"`
 }
+
+// GetBody returns the body of the HTTP request.
+func (h *HTTP) GetBody() string {
+	if h.Body != "" {
+		return h.Body
+	}
+	if h.BodyFrom != nil && h.BodyFrom.Bytes != nil {
+		return string(h.BodyFrom.Bytes)
```

Comment:
@alexec mentioned:

> I think this is probably correct - string would not work because it cannot encode certain characters, I think the JSON encoding is the body in base 64

Does the `string(...)` cast here not have the same problem?

Reply:
I think type-casting will introduce weird issues with character encoding.

Instead, lets have a separate `bodyBytes` and `GetBodyBytes`, that way the users cannot be impacted by it.